interfaces/core.go

```diff
@@ -113,6 +113,13 @@ type Info struct {
 	Slots   []*snap.SlotInfo
 }
 
+type ConnectionInfo struct {
+	Interface string
+	Notes     []string
+	Plug      *snap.PlugInfo
```

Comment:
When would those be nil? Could we use plugRef / slotRef instead?

Reply:
It can be nil if the method is asked to return disconnected plugs/slots as well. In that case the refs would be "".

Reply:
Yeah, this API call is meant to support all the variants of `snap connections` command discussed here: https://forum.snapcraft.io/t/rfc-snap-connections-command/4296/4